mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -37,8 +37,38 @@ private[spark] trait ClassifierParams
       schema: StructType,
       fitting: Boolean,
       featuresDataType: DataType): StructType = {
-    val parentSchema = super.validateAndTransformSchema(schema, fitting, featuresDataType)
-    SchemaUtils.appendColumn(parentSchema, $(rawPredictionCol), new VectorUDT)
+    // TODO: Support casting Array[Double] and Array[Float] to Vector when FeaturesType = Vector
+    SchemaUtils.checkColumnType(schema, $(featuresCol), featuresDataType)
+    if (fitting) {
+      SchemaUtils.checkNumericType(schema, $(labelCol))
+    }
+    val newSchema = SchemaUtils.appendColumn(schema, $(predictionCol), DoubleType,
+      nullable = false, generatePredictionMetadata(schema))
+    SchemaUtils.appendColumn(newSchema, $(rawPredictionCol), new VectorUDT)
+  }
+
+  protected def generatePredictionMetadata(schema: StructType): Metadata = {
+    // The label column for base binary classifier of OneVsRest will not be retained during
```

Comment:
Yes. The annotation is to clarify if the current ```Classifier``` is act as the base binary classifier of OneVsRest estimator(rather than the OneVsRest itself), its label column will not be retained and we should check whether it exists.